.github/workflows/update-documentation.yml

```diff
@@ -0,0 +1,131 @@
+          [[ "$branch" =~ ^ignite-2\.[0-9][0-9](\.[0-9])?$ ]] && branch_check_failed='false'
+          echo "BRANCH_CHECK_FAILED=$branch_check_failed" >> $GITHUB_ENV
+
+      - name: Check inputs
+        if: ${{ env.BRANCH_CHECK_FAILED == 'true' }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('The branch should match the released Ignite since 2.10: ${{ env.IGNITE_BRANCH }}')
+
+      - name: Check out 'ignite' repository code
+        uses: actions/checkout@v3
+        with:
+          repository: 'apache/ignite'
+          ref: ${{ env.IGNITE_BRANCH }}
+
+      - name: Get POM version
+        run: |
+          ver=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+          echo "IGNITE_VERSION=${ver}" >> $GITHUB_ENV
+          
+          version_check_failed='true'
+          [[ "$ver" =~ ^2\.[0-9][0-9]\.[0-9]$ ]] && version_check_failed='false'
+          echo "VERSION_CHECK_FAILED=$version_check_failed" >> $GITHUB_ENV
+
+      - name: Check git version
+        if: ${{ env.VERSION_CHECK_FAILED == 'true' }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('Version should be greater 2.10 and not be a snapshot: ${{ env.IGNITE_VERSION }}')
+
+      - name: Check out 'ignite-website' repository code
+        uses: actions/checkout@v3
+
+      - name: Get latest version
+        run: |
+          latest_version=$(cat ./latest | grep "version" | cut -d'=' -f2)
+          echo "LATEST_VERSION=$latest_version" >> $GITHUB_ENV
+
+      - name: Install prerequisites
+        run: |
+          sudo apt-get install ruby-full build-essential zlib1g-dev
+          echo '# Install Ruby Gems to ~/gems' >> ~/.bashrc
+          echo 'export GEM_HOME="$HOME/gems"' >> ~/.bashrc
+          echo 'export PATH="$HOME/gems/bin:$PATH"' >> ~/.bashrc
+          source ~/.bashrc
+          sudo chown -R $(whoami) /var/lib/gems/*
+          gem install jekyll bundler
+
+      - name: Build ${{ env.IGNITE_VERSION }} documentation
+        run: |
+          cd ${{ github.workspace }}/_docs
+          
+          if [[ "${{ env.IGNITE_VERSION }}" == "${{ env.LATEST_VERSION }}" ]]
+          then
+            echo "Building ${{ env.IGNITE_VERSION }} version (latest)."
+            ./build.sh --version=${{ env.IGNITE_VERSION }} --github-branch=${{ env.IGNITE_BRANCH }} --latest
+          else
+            echo "Building ${{ env.IGNITE_VERSION }} version."
+            ./build.sh --version=${{ env.IGNITE_VERSION }} --github-branch=${{ env.IGNITE_BRANCH }}
+          fi
+
+      - name: Commit changes
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions@github.com
+
+          git pull origin master
+          git add --all
+          git commit -a -m "[bot] Update ${{ env.IGNITE_VERSION }} documentation."
+          git push origin master
```

Comment:
I think we should create a new ticket to the ASF INFRA team for enabling the GITHUB_TOKEN because here it is used implicitly).